utils/python/CIME/XML/entry_id.py

```diff
@@ -56,6 +56,40 @@ def get_type_info(self, vid):
             return None
         else:
             return self._get_type_info(node)
+    
+    def _get_default(self, node):
+        default = self.get_optional_node("default_value", root=node)
+        if default is not None:
+            return default.text
+        else:
+            return None
+        
+    
+    # Get type description , expect child with tag "type" for node         
+    def _get_type (self, node):
+        type_node = self.get_optional_node("type", root=node)
+        if type_node is not None:
```

Comment:
I think this may need to be if type_node is not None and type_node.text is not None?